lib/client/mcp/claude/config.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package claude
+
+import (
+	"encoding/json"
+	"maps"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/gravitational/trace"
+)
+
+// Config represents the Claude Desktop config.
+type Config struct {
+	// MCPServers specifies a list of MCP servers.
+	MCPServers map[string]MCPServer `json:"mcpServers,omitempty"`
+	// AllFields preserves all fields of the config.
+	AllFields map[string]any `json:"-"`
+}
+
+// AddMCPServer adds a new MCP server to the config.
+func (c *Config) AddMCPServer(name string, mcpServer MCPServer) {
+	if c.MCPServers == nil {
+		c.MCPServers = make(map[string]MCPServer)
+	}
+	c.MCPServers[name] = mcpServer
+}
+
+// MarshalJSON implements json.Marshaler.
+func (c Config) MarshalJSON() ([]byte, error) {
+	// Shallow copy.
+	c.AllFields = maps.Clone(c.AllFields)
+	c.updateFields()
+	data, err := json.Marshal(c.AllFields)
+	return data, trace.Wrap(err)
+}
+
+func (c *Config) updateFields() {
+	if c.AllFields == nil {
+		c.AllFields = make(map[string]any)
+	}
+	if len(c.MCPServers) != 0 {
+		c.AllFields["mcpServers"] = c.MCPServers
+	}
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (c *Config) UnmarshalJSON(data []byte) error {
+	type Alias Config
+	if err := json.Unmarshal(data, (*Alias)(c)); err != nil {
+		return trace.Wrap(err)
+	}
+	var allFields map[string]any
+	if err := json.Unmarshal(data, &allFields); err != nil {
```

Comment:
I'm having a hard time understanding what the 2-phase unmarshal is doing here.

Reply:
once for known fields and once for unknown fields. it's quite awkward to do this in golang. i didn't find a good way to do this.

Reply:
let me try json patch

Reply:
I looked into json patch but it doesn't work very well. as suggested by @Tener , i switched to https://github.com/tidwall/sjson

let me know what you think about the latest changes.